api/src/main/java/io/opencensus/trace/Span.java

```diff
@@ -144,8 +144,20 @@ public final void addAnnotation(String description) {
    * higher level applications.
    *
    * @param networkEvent the network event to add.
+   * @deprecated Use {@link #addMessageEvent} for higher level applications.
    */
-  public abstract void addNetworkEvent(NetworkEvent networkEvent);
+  @Deprecated
+  public void addNetworkEvent(NetworkEvent networkEvent) {
+    addMessageEvent(networkEvent.getMessageEvent());
+  }
+
+  /**
+   * Adds a MessageEvent to the {@code Span}.
+   *
+   * @param messageEvent the message to add.
+   * @since 0.10.0
+   */
+  public abstract void addMessageEvent(MessageEvent messageEvent);
```

Comment:
This method should have a default implementation that calls addNetworkEvent, so that it doesn't break any existing subclasses, such as fake Span implementations for tests.

Reply:
Actually it is the opposite way - the `addNetworkEvent` is implemented by `addMessageEvent`.

As to subclasses, I've added implementation for all of them in current version so tests run with same version of api and impl won't break.
If users have older version of opencensus-api library, they won't be able to access this API so everything is fine.

Reply:
Thanks! I understand and fully agree. I initially thought gRPC relies on no-op (but explicitly not depending on opencensus-impl) but turns out I was wrong.